src/hermes/model/merge.py

```diff
@@ -0,0 +1,173 @@
+from hermes.model.path import ContextPath
+
+
+class MergeStrategies:
+    def __init__(self):
+        self._strategies = []
+
+    def select(self, **kwargs):
+        fitting_strategies = [
+            strategy
+            for strategy in self._strategies
+            if strategy.can_handle(kwargs)
+        ]
+        if fitting_strategies:
+            return fitting_strategies[0]
+        else:
+            return None
+
+    def register(self, strategy):
+        self._strategies.append(strategy)
+
+
+class MergeStrategy:
+    @staticmethod
+    def _check_types(item, value):
+        match item:
+            case list(): return any(t in value for t in item)
+            case str(): return item in value
+        return False
+
+    @staticmethod
+    def _check_path(item, value):
+        item = ContextPath.parse(item)
+        value = ContextPath.parse(value)
+        if item == value or item in value:
+            return True
+        return False
+
+    checks = {
+        'type': _check_types,
+        'path': _check_path,
+    }
+
+    def __init__(self, **filter):
+        self._filter = filter
+
+    def _check(self, key, filter, value):
+        if key in filter:
+            check = self.checks.get(key, lambda item, vaue: item in value)
```

Comment:
```suggestion
            check = self.checks.get(key, lambda item, value: item in value)
```

Typo, probably fixing this leads to errors elsewhere.